CDP4Common/Poco/PersonRole.cs

```diff
@@ -42,30 +41,16 @@ public partial class PersonRole
         /// </summary>
         private void PopulatePersonPermissions()
         {
-            var pocos = typeof(Thing).QueryAssembly().GetTypes().Where(x => typeof(Thing).QueryIsAssignableFrom(x));
-            foreach (var poco in pocos)
-            {
-                var field = poco.QueryField("DefaultPersonAccess");
-                if (field == null)
-                {
-                    continue;
-                }
-
-                var accessRight = (PersonAccessRightKind)field.GetRawConstantValue();
-                if (accessRight != PersonAccessRightKind.NONE)
-                {
-                    continue;
-                }
-
-                ClassKind classkind;
-                Enum.TryParse(poco.Name, out classkind);
-                var personPermission = new PersonPermission(Guid.NewGuid(), null, null)
-                                           {
-                                               AccessRight = accessRight,
-                                               ObjectClass = classkind
-                                           };
-                this.PersonPermission.Add(personPermission);
-            }
+            var provider = new DefaultPermissionProvider();
+            provider.GetDefaultTypeNamePersonPermissions()
+                    .Where(x => x.Value.Equals(PersonAccessRightKind.NONE))
+                    .ToList()
+                    .ForEach(x => {
+                        var personPermission = new PersonPermission(Guid.NewGuid(), null, null);
+                        personPermission.AccessRight = x.Value;
+                        personPermission.ObjectClass = (ClassKind)Enum.Parse(ClassKind.GetType(), x.Key);
```

Comment:
Maybe store ClassKind.GetType() in a var out of the ForEach loop, but that's up to you.

Reply:
No problem